### DISCUSSION TITLE
Newest gradle is run despite using wrapper

### DISCUSSION THREAD

**Discussion by jacek-bulatek** [community/community#25913] [Actions]
<p>Hello,<br>
My gradle.yaml:</p>
<pre><code class="lang-auto"># This workflow will build a Java project with Gradle
# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle

name: Java CI with Gradle

on:
  push:
    branches: [ xxx ]
  pull_request:
    branches: [ xxx ]

jobs:
  build:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v2
    - uses: gradle/wrapper-validation-action@v1
    - name: Set up JDK 1.8
      uses: actions/setup-java@v1
      with:
        java-version: 1.8
    - name: Build with Gradle
      run: gradle wrapper --gradle-version=4.6 --distribution-type=all
</code></pre>
<p>yet build fails due to gradle scripts incompability with newest gradle:</p>
<pre><code class="lang-auto">Run gradle wrapper --gradle-version=4.6 --distribution-type=all

Welcome to Gradle 6.5.1!

Here are the highlights of this release:
 - Experimental file-system watching
 - Improved version ordering
 - New samples

For more details see https://docs.gradle.org/6.5.1/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)
Configuration on demand is an incubating feature.

FAILURE: Build failed with an exception.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.

Use '--warning-mode all' to show the individual deprecation warnings.
* Where:
See https://docs.gradle.org/6.5.1/userguide/command_line_interface.html#sec:command_line_warnings
Build file '/home/runner/work/xxx/xxx/xxx/build.gradle' line: 18

* What went wrong:
A problem occurred evaluating project ':desktop'.
&gt; Could not get unknown property 'classesDir' for main classes of type org.gradle.api.internal.tasks.DefaultSourceSetOutput.
</code></pre>
<p>I am new to whole gradle thing, so perhaps I am doing something wrong?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/jacek-bulatek">@jacek-bulatek</a>,</p>
<p>Glad to see you in Github Forum ticket!</p>
<p>When you perform the <code>wrapper</code> task using  <code>gradlew</code>, it will still execute with current version (Gradle  <code>6.5.1</code> ). The arguments only change the values that are written to  <code>gradle-wrapper.properties</code> . The  <code>gradle-wrapper.jar</code>  file is still the one copied from the  <code>6.5.1</code>  distribution. The next build run using  <code>gradlew</code>  will download the new version distribution and use the new version(version <code>4.6</code>).</p>
<p>I copied your code and confirmed it on my workflow: <a href="https://github.com/weide-zhou/protobuf-gradle-plugin/runs/939173587?check_suite_focus=true#step:6:10" rel="nofollow noopener">https://github.com/weide-zhou/protobuf-gradle-plugin/runs/939173587?check_suite_focus=true#step:6:10</a></p>
<p><img src="https://user-images.githubusercontent.com/52772761/181094080-d06c04f2-45c3-4bc7-9d04-dddab045988a.png" alt="image" data-base62-sha1="fJaaNoGBN5NGtmjlpdKM4TMm0iy" width="677" height="352"></p>
<p>Please refer to this link <a href="https://discuss.gradle.org/t/gradle-wrapper-jar-not-updated-from-6-0-1-to-6-3/35455/4" rel="nofollow noopener">here</a> for more details.</p>
<p>Your error should be due to the deprecated <code>classesDir</code>  property, please use getClassesDirs() instead.</p>
<p>Please refer to below link for more details:<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://aws1.discourse-cdn.com/gradle/optimized/2X/9/97278102a4b39e2bc437f90f769c3310f492900f_2_32x32.ico" class="site-icon" width="16" height="16">
      <a href="https://discuss.gradle.org/t/could-not-get-unknown-property-classesdir-for-main-classes-of-type-org-gradle-api-internal-tasks-defaultsourcesetoutput/30570" target="_blank" rel="nofollow noopener" title="10:46PM - 13 February 2019">Gradle Forums – 13 Feb 19</a>
  </header>
  <article class="onebox-body">
    <img src="https://aws1.discourse-cdn.com/gradle/original/2X/a/a9c8968f1d3907dbb4d26c83913a39ab36931984.png" class="thumbnail onebox-avatar" width="60" height="60">

<h3><a href="https://discuss.gradle.org/t/could-not-get-unknown-property-classesdir-for-main-classes-of-type-org-gradle-api-internal-tasks-defaultsourcesetoutput/30570" target="_blank" rel="nofollow noopener">Could not get unknown property 'classesDir' for main classes of type...</a></h3>

<p>I’m trying to bump https://github.com/apache/fineract from a (very, I know) old Gradle 2.10 to current 5.2.1 in https://github.com/apache/fineract/compare/develop...vorburger:bump-gradle-wrapper for https://issues.apache.org/jira/browse/FINERACT-700...</p>

  <p><span class="label1">Reading time: 1 mins 🕑</span>
    <span class="label2">Likes: 1 ❤</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Thanks.</p>

**Comment by jacek-bulatek** [comment, marked as answer]
<p>Creating and using gradlew helped.</p>
